schema/pgsql/schema.sql

```diff
@@ -166,7 +166,7 @@ BEGIN
     END IF;
   END LOOP;
 
-  RETURN 100 * (total_time - problem_time) / total_time;
+  RETURN 100 * CAST(total_time - problem_time AS decimal) / total_time;
```

Comment:
The function already uses the PostgreSQL-specific cast syntax (`NULL::tinyuint`), I'd keep this consistent here.